addons/core/functions/fnc_processPlayerPositions.sqf

```diff
@@ -18,80 +18,84 @@
     Example:
         call TFAR_fnc_processPlayerPositions;
 */
-if (getClientStateNumber != 10) exitWith {"BI HAS CRAPPY WEIRD BUGS U KNOW! (Keeps PFH from firing after server disconnect)"};
-//if (isNull (findDisplay 46)) exitWith {};
-//if (isNull TFAR_currentUnit) exitWith {};
+if (getClientStateNumber != 10) exitWith {};
+
 private _startTime = diag_tickTime;
 
 //Process queued Near Players
 if (!TFAR_currentNearPlayersProcessed) then {
+
+    if (TFAR_currentNearPlayersProcessing isEqualTo []) exitWith {TFAR_currentNearPlayersProcessed = true};
+
     private _nearPlayersCount = count TFAR_currentNearPlayersProcessing;
-    if (_nearPlayersCount == 0) exitWith {TFAR_currentNearPlayersProcessed = true};
-    private _playersToProcess = _nearPlayersCount min 30;//Plugin POS info takes about 10 microseconds meaning 10 position updates block for 0.1 ms
+
+    private _playersToProcess = _nearPlayersCount min 30; //Plugin POS info takes about 10 microseconds meaning 10 position updates block for 0.1 ms
     if (_playersToProcess == 0) exitWith {TFAR_currentNearPlayersProcessed = true};
 
     {
         private _controlled = _x getVariable ["TFAR_controlledUnit", objNull];
         private _unitName = name _x;
-        if (_x getVariable ["TFAR_forceSpectator",false]) then {
-            _unitName = _x getVariable ["TFAR_spectatorName",_unitName];
+        if (_x getVariable ["TFAR_forceSpectator", false]) then {
+            _unitName = _x getVariable ["TFAR_spectatorName", _unitName];
         };
         if (isNull _controlled) then {
             [_x, true, _unitName] call PROFCONTEXT_NORTN(TFAR_fnc_sendPlayerInfo);
         } else {
             [_controlled, true, _unitName] call PROFCONTEXT_NORTN(TFAR_fnc_sendPlayerInfo);
         };
-    } forEach (TFAR_currentNearPlayersProcessing select [0,_playersToProcess]); //commy2
-
-    //private _startIndex = _nearPlayersCount - _playersToProcess; //Is min 0
-    //for "_y" from _startIndex to _nearPlayersCount-1 step 1 do {
-    //    private _unit = (TFAR_currentNearPlayersProcessing select _y);
-    //};
+        nil
+    } count (TFAR_currentNearPlayersProcessing select [0, _playersToProcess]); //commy2
```

Comment:
count+extra command can be slower than a simple forEach.
nil isn't needed here because it would use the returned value from sendPlayerInfo

Reply:
`count` is always faster then `foreach`.
no `_forEachIndex` every iteration -> it's faster

Reply:
Yes. But if you add an additional commands thats 3 microseconds slower on each iteration. versus the about 1 microsecond that count is faster.